### PR TITLE
Extend tedge-agent with config and log management

### PR DESCRIPTION
**Pull request by albinsuresh**
## Proposed changes

Instead of creating a new `tedge-device-management` plugin with the additional features, the existing agent itself was extended with the newer feature, guarded behind a `v1` flag. The extended agent is installed as `/usr/bin/tedge-agent-v1` along with the existing `tedge-agent` and can be spawned from there, after stopping the running `tedge-agent` instance.

No system tests added for this specific variant as existing system tests for the dedicated operation plugins will be updated later by removing those plugins and using the updated agent instead.

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Improvement (general improvements like code refactoring that doesn't explicitly fix a bug or add any new functionality)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Paste Link to the issue

https://github.com/thin-edge/thin-edge.io/issues/2433

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [x] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [x] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Review by didier-wenzek — COMMENTED** [review]
`tedge-agent` support for log and configuration is working as expected.

However:

- As discussed offline, it would be better to use the symlink approach rather than a `--v2` option. I.e. to have a tedge-agent-v2 symlink to tedge which would include all these plugins.
- The `tedge-log-plugin` and `tedge-configuration-plugin` are somehow included twice as also wrapped by the `tedge` multi-command.
- One needs to consider removing the use of `sudo` under the hood, as `tedge-agent` must then be launched as `root`. This will have to be done only when fully ready to deprecate the current packaging.

File: crates/core/tedge_agent/src/lib.rs [hunk 1]
```diff
@@ -20,13 +20,13 @@ use tedge_config::system_services::set_log_level;
...
-mod state_repository;
-mod tedge_operation_converter;
-mod tedge_to_te_converter;
+pub mod agent;
+pub mod file_transfer_server;
+pub mod restart_manager;
+pub mod software_manager;
+pub mod state_repository;
+pub mod tedge_operation_converter;
+pub mod tedge_to_te_converter;
```
[the remark this answers is not in the document]
Comment (albinsuresh) [reply crates/core/tedge_agent/src/lib.rs:29]: Yes. It was a side effect of my first approach creating an additional `tedge-agent-v2` binary from the same project.

**Comment by albinsuresh** [comment]
> * As discussed offline, it would be better to use the symlink approach rather than a `--v2` option. I.e. to have a tedge-agent-v2 symlink to tedge which would include all these plugins.

Yes.

> * The `tedge-log-plugin` and `tedge-configuration-plugin` are somehow included twice as also wrapped by the `tedge` multi-command.

Yes, those two plugins would be removed once the migration to the new agent is complete. We're keeping it just for the transition phase.

> * One needs to consider removing the use of `sudo` under the hood, as `tedge-agent` must then be launched as `root`. This will have to be done only when fully ready to deprecate the current packaging.

To be discussed.

**Review by didier-wenzek — APPROVED** [review]
Approved. Nice to see this is just a matter of moving around actor builders.

**Comment by albinsuresh** [comment]
> Approved. Nice to see this is just a matter to move around actor builders.

Yeah, it was a bit of a surprise to me too, even though it was clear that this shouldn't be that hard, based on the `c8y-device-management` plugin experience. But, I still didn't expect it to be so trivial. Glad to see that we did a pretty good job with the actor based architecture, making work like this so much easier.

**Review by reubenmiller — APPROVED** [review]
Approved

**Pull request merged**